docs/using.rst

```diff
@@ -24,6 +24,68 @@ Although gateway token is very handy during development or testing, it's
 recommended to use this token with care because it can be used to
 authenticate any device.
 
+Authentication via Certificates
+-------------------------------
+
+As can be seen in the system configuration settings of hawkBit, there is a
+third option to authenticate the targets. An "Allow targets to authenticate via
+a certificate authenticated by a reverse proxy" option. To use this
+authentication method a TLS reverse proxy server needs to be set up.
+The client and reverse proxy server perform a "SSL-handshake" that means the
+client validates the server certificate of the reverse proxy server with its
+list of trusted certificates.
+
+The clients request has:
+
+- to have a TLS connection to the reverse proxy server
+- to contain the client certificate
+- to have the common name of the server certificate match the server
+  name set in the configuration file as "hawkbit_server"
+
+The purpose of the reverse proxy is to:
+
+- disband the TLS connection
+- check if sent client certificate is valid
+- extract the common name and fingerprint of the client certificate
+- forward the common name and fingerprint as HTTP headers to the
+  hawkBit server
+
+When the hawkBit server receives the request it checks if:
+
+- sent common name matches with the controller ID of the target
+- sent fingerprint(s) matches the expected fingerprint(s) which is set
+  in the system configuration settings of hawkBit
+
+The client certificate will only be used if no tokens are set and a valid path
+to a certificate and its key is given in the configuration file.
+
+Here an example of how the configuration file might look like:
+
+[client]
+hawkbit_server            = CN_server_certificate:443
+ssl                       = true
+ssl_verify                = true
```

Comment:
From https://github.com/rauc/rauc-hawkbit-updater/pull/38#discussion_r651111076:
> What happens if these options are false?

Reply:
I added the answer here : https://github.com/flobz/rauc-hawkbit-updater/blob/419925f13ac05d311d92e9129661783456726d78/docs/using.rst?plain=1#L66
If the CA of the reverse proxy server is untrusted set ``ssl_verify`` to ``false``.

Reply:
Okay, but what about the `ssl` option? That is required, right? Should be documented and a run time check would also be nice.

Reply:
you're right I forgot this check : a059fb3